### PR TITLE
Hyphenate 'pointer-to-member' when it is an adjective.

### LINKED ISSUE #1369: "pointer to member" vs "pointer-to-member"
Pick one.

e.g. just in overloading.tex:

````
overloading.tex:a pointer to member, or \tcode{std::nullptr_t}
overloading.tex:pointer-to-member-function type.
overloading.tex:pointer to member function type.
overloading.tex:function name is required to have the form of a pointer to member as
overloading.tex:                                // pointer to member
overloading.tex:For every pointer to member type \tcode{\placeholder{T}} or type \tcode{std::nullptr_t} there
overloading.tex:is an enumeration or pointer to member type,
overloading.tex:is a pointer, pointer-to-member, or scoped enumeration type, there exist candidate operator
````

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #1369.

### CODE REVIEW

**Comment by tkoeppe** [comment]
Looks good, will leave to @zygoloid to merge.

File: source/conversions.tex [hunk 1]
```diff
@@ -524,21 +524,21 @@
 object. The null pointer value is converted to the null pointer value of
 the destination type.
 
-\rSec1[conv.mem]{Pointer to member conversions}
+\rSec1[conv.mem]{Pointer-to-member conversions}
```
Comment (zygoloid) [inline source/conversions.tex]: I'm concerned that this change suggests a false parallel: previously, all the "X-to-Y conversion"s converted an X into a Y. A pointer to member conversion definitely does not convert a pointer into a member. I'm also still trying to decide if the rules for adjectives really apply here; "pointer to member type" seems more like "office chair store" (which would not be written as "office-chair store") than like, say, "office-themed store".

We have quite a few other cases (such as "pointer to function types") that follow a similar non-hyphenation pattern. These seem quite natural to me as-is.

Comment (jensmaurer) [inline source/conversions.tex]: We have 15 appearances where "pointer-to-member" is, in fact, hyphenated, so I presume you would accept a patch that removes the hyphens there?

Comment (zygoloid) [inline source/conversions.tex]: Let's discuss this at Kona and come up with a rule, then apply it consistently.

Comment (tkoeppe) [reply source/conversions.tex]: We decided that we can get over this. Fine as written.

**Comment by tkoeppe** [comment]
I stole this commit, rebased it, added the definition of "pointer-to-member type", and committed this as cbd25c4b3e6f83aaf2fa9fe894ff52b1f428a8c5.

**Pull request closed without merging**